pkg/apis/core/fuzzer/fuzzer.go

```diff
@@ -309,6 +309,23 @@ var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
 			c.FuzzNoCustom(ct)                                          // fuzz self without calling this function again
 			ct.TerminationMessagePath = "/" + ct.TerminationMessagePath // Must be non-empty
 			ct.TerminationMessagePolicy = "File"
+			// Match defaulting in pkg/apis/core/v1/defaults.go.
+			_, hasCPUReq := ct.Resources.Requests[core.ResourceCPU]
+			_, hasCPULim := ct.Resources.Limits[core.ResourceCPU]
+			_, hasMemReq := ct.Resources.Requests[core.ResourceMemory]
+			_, hasMemLim := ct.Resources.Limits[core.ResourceMemory]
+			if hasCPUReq || hasCPULim {
+				ct.ResizePolicy = append(ct.ResizePolicy, core.ContainerResizePolicy{
```

Comment:
if SetDefaults_Pod is defaulting new fields on container when this feature is enabled, won't that instant-restart pods on API server upgrade (https://github.com/kubernetes/kubernetes/issues/122028#issuecomment-1837528357) until the oldest node is 1.31 and includes the fix in https://github.com/kubernetes/kubernetes/pull/124220?

Reply:
possible options to avoid the instant restart were to wait for 1.31 to be the oldest supported skewed node or to move the "defaulting" of the pod fields to PrepareForCreate (https://github.com/kubernetes/kubernetes/issues/122028#issuecomment-1838905110)

Reply:
We should just remove the defaulting... Kubelet already handles the missing resize policy as an implicit default. I don't see any reason to set the explicit default.

Reply:
I haven't followed all the discussions on the feature / API, but that seems likely to be the best option